src/sidebar/services/tags.js

```diff
@@ -23,13 +23,23 @@ function tags(localStorage) {
    * Return a list of tag suggestions matching `query`.
    *
    * @param {string} query
+   * @param {number} limit - Optional limit of the results.
    * @return {Tag[]} List of matching tags
    */
-  function filter(query) {
+  function filter(query, limit = 0) {
+    const limitNumber = Number(limit); // ensures we are dealing with a number
```

Comment:
I think I understand what you're trying to do here. In general however I think that it is often preferable to use a different type of value (`null` or `undefined`) to indicate that a value is not set rather than a magic value of the same type (`0`). The reason being that it reduces the chance of subtle bugs where the special value (`0`) is manipulated without checking it first. If this happens with `null` it will crash loudly. It also avoids scenarios where the behavior of the function suddenly changes if a programatically generated value happens to equal the special value.

My preference here would be to use `null` as before to indicate "no limit" but test for that value specifically below (ie. `limit === null`).